### PR TITLE
Typecast refactoring

### PR DESCRIPTION
**Pull request by Tigrov**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  |

### CODE REVIEW

**Review by terabytesoftw — APPROVED** [review]

File: src/Schema.php [hunk 1]
```diff
@@ -493,21 +493,41 @@ protected function loadColumnSchema(array $info): ColumnSchemaInterface
...
+     */
+    private function normalizeDefaultValue(?string $defaultValue, ColumnSchemaInterface $column): mixed
+    {
+        return match (true) {
+            $defaultValue === null,
+            $defaultValue === 'null',
+            $defaultValue === '',
+            $column->isPrimaryKey()
+                => null,
+            in_array($defaultValue, [
```
Comment (samdark) [inline src/Schema.php:519]: Three conditions or `is_array`? It's better to use the same in both cases.

Comment (Tigrov) [reply src/Schema.php]: Done, thanks.

File: src/Schema.php [hunk 2]
```diff
@@ -493,19 +493,32 @@ protected function loadColumnSchema(array $info): ColumnSchemaInterface
...
+    private function normalizeDefaultValue(?string $defaultValue, ColumnSchemaInterface $column): mixed
+    {
+        if ($column->isPrimaryKey()) {
+            return null;
         }
 
-        return $column;
+        return match ($defaultValue) {
+            null, 'null', '' => null,
+            'CURRENT_TIMESTAMP', 'CURRENT_DATE', 'CURRENT_TIME' => new Expression($defaultValue),
```
Comment (darkdef) [inline src/Schema.php:519]: Need add check for type of column. Without check type of column - with default value in column equal 'CURRENT_TIMESTAMP' (string), default value will be incorrect 
For this case (example):
create table TABLE_NAME
(
    COLUMN_1 VARCHAR2(100) default current_timestamp
)

Comment (Tigrov) [reply src/Schema.php:519]: There is no special date type `timestamp` or `datetime` in SQlite.
https://www.sqlite.org/datatype3.html

For this purpose `text` type can be used.

Comment (darkdef) [reply src/Schema.php:519]: https://github.com/yiisoft/db-sqlite/blob/master/tests/Provider/SchemaProvider.php#L162

Comment (darkdef) [reply src/Schema.php:519]: We have dbtype in schema of column, and need check type in condition

**Review by darkdef — CHANGES_REQUESTED** [review]

File: src/Schema.php (hunk 2 above)
Comment (Tigrov) [reply src/Schema.php:519]: Added test with `STRICT` table where `TIMESTAMP` type is not allowed but `TEXT` type for storing date and time is okey.
```sql
CREATE TABLE "timestamp_default" (
  id INTEGER PRIMARY KEY,
  text_col TEXT NOT NULL DEFAULT 'CURRENT_TIMESTAMP',
  timestamp_text TEXT NOT NULL DEFAULT CURRENT_TIMESTAMP
) STRICT;
```

Could you show a test where current changes without `check for type of column` will fail?

Comment (darkdef) [reply src/Schema.php:519]: test will be more reliable. Simply add test for column text_col with assert: defaultValue not instaceof Expression

Comment (darkdef) [reply src/Schema.php:519]: The current test is sufficient. Make it work without errors and that's it. Thank you

**Review by darkdef — APPROVED** [review]
Thanks. Please add line to changelog

**Pull request merged**